### PR TITLE
added pack unpack bin2num function support Bigint

### PR DESCRIPTION
**Pull request by baryon**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -237,13 +237,17 @@ export function getPreimage(tx, inputLockingScriptASM: string, inputAmount: numb
 // Converts a number into a sign-magnitude representation of certain size as a string
 // Throws if the number cannot be accommodated
 // Often used to append numbers to OP_RETURN, which are read in contracts
-// TODO: handle bigint
 export function num2bin(n: number, dataLen: number): string {
+  return pack(n, dataLen);
```
Comment (xhliu) [inline src/utils.ts:241]: Put `pack()` body into num2bin and remove `pack()` since they are the same. Same for `unpack`.

File: src/utils.ts [hunk 2]
```diff
@@ -237,13 +237,17 @@ export function getPreimage(tx, inputLockingScriptASM: string, inputAmount: numb
...
+}
+
+//Support Bigint
+export function pack(n: any, dataLen: number): string {
```
Comment (xhliu) [inline src/utils.ts:245]: Use `n : number | BigInt`
https://github.com/scrypt-sv/testing/blob/145d64703e7edb1602800043915cf77e861b4dd8/src/local.ts#L20-L28

**Review by xhliu — CHANGES_REQUESTED** [review]
Looking good, well done!

Just one thing: switch to BigInt type.

**Comment by baryon** [comment]
Done

**Review by xhliu — APPROVED** [review]
👍

**Pull request merged**